src/constraint.rs

```diff
@@ -416,7 +432,120 @@ impl OrderedElementsConstraint {
 
 impl ConstraintValidator for OrderedElementsConstraint {
     fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
```

Comment:
This method is long enough that I had to do a lot of scrolling up/down to read it. Consider breaking some sections out into helper methods.